### DISCUSSION TITLE
Exported ONNX model

### DISCUSSION THREAD

**Discussion by bschmer** [ultralytics/yolov5#2170] [Q&A]
Sorry if this is a FAQ.  I've poked around quite a bit and haven't found an answer.  Newbie to YOLO, pytorch, OONX, etc so please forgive any misphrased questions.

I've trained a YOLOv5s model and it works great with the detect.py that's included with YOLO.  When I export the model to ONNX and try to run inference using the model in pytorch, the shape and content of the output is extremely different from the output produced by detect.py.  The reason I'd like to figure out the ONNX model is that I have an existing pipeline that does further processing on the ONNX model output, but it doesn't work properly due to the output differences.

I've tried to make sure that the versions of all components used by YOLO are the same ones used by the additional pipeline, but that hasn't helped at all.

Any suggestions?

**Comment by glenn-jocher** [comment]
@bschmer try setting this line to False before exporting, and then visualizing with Netron to see if the outputs are correct:
https://github.com/ultralytics/yolov5/blob/ace3e02e406307ed0cebc25aebe15835767107b2/models/export.py#L53

**Reply by bschmer** [reply]
That seems to have done the trick.  Thanks!
